src/assets/scripts/base/PositionModel.js

```diff
@@ -238,15 +204,13 @@ export default class PositionModel {
     }
 
     /**
-     * Fascade for `adjustForMagneticNorth`
-     *
-     * Adjusts x & y coordinates from true north to magnetic north
-     *
+     * Determine the `x` and `y` values of the `PositionModel`, used for drawing on the canvas
      * @for PositionModel
-     * @method adjustXYForMagneticNorth
+     * @method _calculateScreenPosition
+     * @private
      */
-    adjustXYForMagneticNorth() {
-        const { x, y } = adjustForMagneticNorth(this.x, this.y, this.magnetic_north);
+    _calculateScreenPosition() {
+        const [x, y] = PositionModel.calculatePosition(this.gps, this.reference_position, this.magnetic_north);
```

Comment:
This call to `PositionModel.calculatePosition()` fails when its the airport itself trying to create its `PositionModel`. Since there is no reference position (everything will be in reference to the airport), the method throws the error in the first line:
```javascript
    if (!coordinates || !referencePostion || !magneticNorth) {
        throw new TypeError('Invalid parameter. PositionModel.getPosition() requires coordinates, referencePostion ' +
            'and magneticNorth as parameters');
    }
```

Any ideas on how this might best be resolved?